packages/ui/src/components/TextLink/index.tsx

```diff
@@ -17,6 +21,33 @@ export type Props = AnchorHTMLAttributes<HTMLAnchorElement> & {
 
 const TextLink = ({ className, children, text, icon, type = 'primary', to, ...rest }: Props) => {
   const { t } = useTranslation();
+  const { isMobile } = usePlatform();
+  const { setModalState } = useIframeModal();
+
+  // By default the behavior of opening a new window is not supported in WkWebView, or in android webview.
+  // Hijack the hyperlink props and open the link in an iframe modal instead.
+
+  const hyperLinkProps = useMemo(() => {
```

Comment:
it looks like no need to check outside `onClick`?

Reply:
updated